File: my-folder/0240-search-a-2d-matrix-ii/solution.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int n = matrix[0].size()-1;
        int m = 0;
        while(n>=0 && m<matrix.size() ){
            if(matrix[m][n]==target) return true;
            if(matrix[m][n]<target) m++;
            else n--;
        }
        return false;
    }
};

File: my-folder/0240-search-a-2d-matrix-ii/solution.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int row = 0;
        int col = matrix[0].length-1;
        while(row<matrix.length && col>=0){
            int t = matrix[row][col];
            if(t==target) return true;
            else if(t<target) row++;
            else col--;
        }
        return false;
        
    }
}
